onnxruntime/contrib_ops/cuda/layer_norm.cc

```diff
@@ -64,8 +64,8 @@ Status LayerNorm<T, U, simplified>::ComputeInternal(OpKernelContext* ctx) const
   const TensorShape& x_shape = X->Shape();
   const int64_t axis = HandleNegativeAxis(axis_, x_shape.NumDimensions());
 
-  auto n1 = x_shape.SizeToDimension(axis);
-  auto n2 = x_shape.SizeFromDimension(axis);
+  int n1 = gsl::narrow<int>(x_shape.SizeToDimension(axis));
```

Comment:
>int n1 [](start = 2, length = 6)

On windows 32-bit signed is potentially too small. Do we want to change the type for it?

Reply:
Then in debug build, it will throw. In release build, such errors will be ignored. 
Suggestions?